tests/integration/devfile/cmd_devfile_url_test.go

```diff
@@ -29,388 +29,453 @@ var _ = Describe("odo devfile url command tests", func() {
-			stdout = helper.Cmd("odo", "push", "--context", subFolderContext).ShouldPass().Out()
-			helper.MatchAllInOutput(stdout, []string{url1 + "." + host, url2})
-
-			stdout = helper.Cmd("odo", "url", "list", "--context", subFolderContext).ShouldPass().Out()
-			helper.MatchAllInOutput(stdout, []string{url1, url2, "Pushed", "false", "ingress"})
-		})
-
-		It("should list ingress url with appropriate state", func() {
-			url1 := helper.RandString(5)
-			url2 := helper.RandString(5)
-			host := helper.RandString(5) + ".com"
+	When("creating a Nodejs component", func() {
+		url1 := helper.RandString(5)
+		host := helper.RandString(5) + ".com"
+		stdout := ""
 
+		BeforeEach(func() {
 			helper.Cmd("odo", "create", "nodejs", "--project", commonVar.Project, componentName).ShouldPass()
-
 			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), commonVar.Context)
 			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile.yaml"), filepath.Join(commonVar.Context, "devfile.yaml"))
-
-			helper.Cmd("odo", "url", "create", url1, "--port", "9090", "--host", host, "--secure", "--ingress").ShouldPass()
-			helper.Cmd("odo", "push").ShouldPass()
-			stdout := helper.Cmd("odo", "url", "list").ShouldPass().Out()
-			helper.MatchAllInOutput(stdout, []string{url1, "Pushed", "true", "ingress"})
-
-			helper.Cmd("odo", "url", "delete", url1, "-f").ShouldPass()
-			helper.Cmd("odo", "url", "create", url2, "--port", "8080", "--host", host, "--ingress").ShouldPass()
-			stdout = helper.Cmd("odo", "url", "list").ShouldPass().Out()
-			helper.MatchAllInOutput(stdout, []string{url1, "Locally Deleted", "true", "ingress"})
-			helper.MatchAllInOutput(stdout, []string{url2, "Not Pushed", "false", "ingress"})
 		})
+		When("creating ingress url1 with port flag and doing odo push", func() {
+			url2 := "nodejs-project-3000-" + helper.RandString(5)
+			BeforeEach(func() {
+				helper.Cmd("odo", "url", "create", url1, "--port", "9090", "--host", host, "--secure", "--ingress").ShouldPass()
+				helper.Cmd("odo", "push").ShouldPass()
+			})
+			When("creating ingress url2 with port flag and doing odo push", func() {
+				BeforeEach(func() {
+					helper.Cmd("odo", "url", "create", url2, "--port", "8080", "--host", host, "--ingress").ShouldPass()
+					stdout = helper.Cmd("odo", "url", "list").ShouldPass().Out()
+				})
+				When("deleting url1", func() {
+					BeforeEach(func() {
+						helper.Cmd("odo", "url", "delete", url1, "-f").ShouldPass()
+						stdout = helper.Cmd("odo", "url", "list").ShouldPass().Out()
+					})
+					It("should check status of url1 and url2", func() {
+						helper.MatchAllInOutput(stdout, []string{url1, "Locally Deleted", "true", "ingress"})
+						helper.MatchAllInOutput(stdout, []string{url2, "Not Pushed", "false", "ingress"})
+					})
+				})
+				It("should check state of url2", func() {
```

Comment:
For readibility, please place the `It`s before the `When`s

Reply:
understood